kornia/geometry/epipolar/essential.py

```diff
@@ -19,6 +21,183 @@
     "relative_camera_motion",
 ]
 
+# Reference
+# Nistér, David. An efficient solution to the five-point relative pose problem. 2004.
+# https://github.com/danini/graph-cut-ransac/blob/master/src/pygcransac/include/estimators/solver_essential_matrix_five_point_nister.h
+# Wei T, Patel Y, Matas J, Barath D. Generalized differentiable RANSAC[J]. arXiv preprint arXiv:2212.13185, 2023.
+# https://github.com/weitong8591/differentiable_ransac/blob/main/estimators/essential_matrix_estimator_nister.py
+
+
+def run_5point(points1: torch.Tensor, points2: torch.Tensor, weights: Optional[torch.Tensor] = None) -> torch.Tensor:
+    r"""Compute the essential matrix using the 5-point algorithm from Nister.
+
+    The linear system is solved by Nister's 5-point algorithm.
+
+    Args:
+        points1: A set of carlibrated points in the first image with a tensor shape :math:`(B, N, 2), N>=8`.
+        points2: A set of points in the second image with a tensor shape :math:`(B, N, 2), N>=8`.
+        weights: Tensor containing the weights per point correspondence with a shape of :math:`(B, N)`.
+
+    Returns:
+        the computed fundamental matrix with shape :math:`(B, 3, 3)`.
+    """
+
+    if (
```

Comment:
use the `KORNIA_CHECK` api

Reply:
Thanks! Should I do it in this way? I changed them to reduce the number of ifs. I will change it 
```
if points1.shape != points2.shape:
        raise AssertionError(points1.shape, points2.shape)
if points1.shape[1] < 5:
        raise AssertionError(points1.shape)
if weights is not None:
        if not (len(weights.shape) == 2 and weights.shape[1] == points1.shape[1]):
            raise AssertionError(weights.shape)

```